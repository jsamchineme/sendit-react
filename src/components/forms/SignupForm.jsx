import React from 'react'
import { Field, reduxForm } from 'redux-form';
import { NavLink } from 'react-router-dom';
import { userSignupRules } from './rules/formRules';


const renderField = ({
  input,
  label,
  type,
  meta: { touched, error, warning }
}) => {
  
  return (
    <div>
      <label>{label}</label>
      {touched &&
        ((error && <div className="error-box active">{error}</div>))}
      <div>
        <input {...input} placeholder={label} type={type} />
      </div>
    </div>
  )
}


const SignupForm = props => {
  const { handleSubmit, serverErrorMessage, pristine, reset, submitting } = props;

  return (
    <form onSubmit={handleSubmit}>
      <div className="form-header">
        Signup
        {serverErrorMessage &&
        <div id="server-error-box" className="error-box active">{serverErrorMessage}</div>}
      </div>
      <div className="form-body">
        <div className="input-group transparent-box">
          <Field
            name="firstname"
            type="firstname"
            component={renderField}
            label="Firstname"
            validate={userSignupRules.firstname}
          />
          <Field
            name="lastname"
            type="lastname"
            component={renderField}
            label="Lastname"
            validate={userSignupRules.lastname}
          />
          <Field
            name="email"
            type="email"
            component={renderField}
            label="your email"
            validate={userSignupRules.email}
          />
          <Field
            name="username"
            type="username"
            component={renderField}
            label="username"
            validate={userSignupRules.username}
          />
          <Field
            name="password"
            type="password"
            component={renderField}
            label="secure password"
            validate={userSignupRules.password}
          />
        </div>
        <div className="input-footer" />
        <div className="row text-center">
          <div className="column">
            <button className="btn submit">Signup</button>
          </div>
          <div className="actions column text-center">
            <div className="v-gap-2" />
            <div>
              <NavLink to='/login'>Login</NavLink>
            </div>
          </div>
        </div>
      </div>
    </form>
  )
}

export default reduxForm({
  form: 'signupForm'
})(SignupForm)