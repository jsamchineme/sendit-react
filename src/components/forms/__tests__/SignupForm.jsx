import React from 'react';
import { shallow } from 'enzyme';
import { SignupForm } from '../SignupForm';
import configureStore from '../../../redux/configureStore';

const store = configureStore();

describe('Tests for signup form', () => {
  it('should render the signup form', () => {
    const props = {
      handleSubmit: jest.fn(), 
      serverErrorMessage: '',
    }
    const wrapper = shallow(<SignupForm  {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});