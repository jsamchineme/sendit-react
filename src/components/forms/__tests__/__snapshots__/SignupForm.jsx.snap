// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for signup form should render the signup form 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <div
    className="form-header"
  >
    Signup
  </div>
  <div
    className="form-body"
  >
    <div
      className="input-group transparent-box"
    >
      <Field
        component={[Function]}
        label="Firstname"
        name="firstname"
        type="firstname"
        validate={
          Array [
            [Function],
            [Function],
          ]
        }
      />
      <Field
        component={[Function]}
        label="Lastname"
        name="lastname"
        type="lastname"
        validate={
          Array [
            [Function],
            [Function],
          ]
        }
      />
      <Field
        component={[Function]}
        label="your email"
        name="email"
        type="email"
        validate={
          Array [
            [Function],
            [Function],
            [Function],
            [Function],
          ]
        }
      />
      <Field
        component={[Function]}
        label="username"
        name="username"
        type="username"
        validate={
          Array [
            [Function],
            [Function],
            [Function],
          ]
        }
      />
      <Field
        component={[Function]}
        label="secure password"
        name="password"
        type="password"
        validate={
          Array [
            [Function],
            [Function],
            [Function],
          ]
        }
      />
    </div>
    <div
      className="input-footer"
    />
    <div
      className="row text-center"
    >
      <div
        className="column"
      >
        <button
          className="btn submit"
        >
          Signup
        </button>
      </div>
      <div
        className="actions column text-center"
      >
        <div
          className="v-gap-2"
        />
        <div>
          <NavLink
            to="/login"
          >
            Login
          </NavLink>
        </div>
      </div>
    </div>
  </div>
</form>
`;
